src/renderer/browser/actions/makeTestInvokeActions.js

```diff
@@ -1,39 +1,51 @@
 import { createActions } from 'spunky';
-import { rpc, wallet } from '@cityofzion/neon-js';
+import { rpc } from '@cityofzion/neon-js';
 import { isArray } from 'lodash';
 
 import getRPCEndpoint from 'util/getRPCEndpoint';
 
 import createScript from 'shared/util/createScript';
 
 import generateDAppActionId from './generateDAppActionId';
+import createArrayScript from '../util/createArrayScript';
+import validateScriptArgs from '../util/validateScriptArgs';
 
 export const ID = 'testInvoke';
 
-const testInvoke = async ({ net, scriptHash, operation, args, encodeArgs }) => {
-  if (!wallet.isScriptHash(scriptHash)) {
-    throw new Error(`Invalid script hash: "${scriptHash}"`);
-  }
-
-  if (typeof operation !== 'string') {
-    throw new Error(`Invalid operation: "${operation}"`);
-  }
-
-  if (!isArray(args)) {
-    throw new Error(`Invalid arguments: "${args}"`);
+const testInvoke = async ({ net, scriptHash, operation, args, script, encodeArgs }) => {
+  let invokeScript;
+
+  if (scriptHash && operation && args) {
+    validateScriptArgs({ scriptHash, operation, args });
+    invokeScript = createScript(scriptHash, operation, args, encodeArgs);
+  } else if (script) {
+    if (typeof script !== 'string' || !isArray(script)) {
```

Comment:
Pretty sure you want `&&` here, not `||`.

Reply:
Fixed this!